### PR TITLE
Adds podman compatibility in makefile

### LINKED ISSUE #1298: Add podman compatibility to Makefile
I would like to use podman for building the argocd operator instead of docker, without breaking it for people who still use docker.

**Is your feature request related to a problem? Please describe.**
I'm using Fedora and the Podman Buildah stack, so i have no Docker installed, so per default i can't build the operator

**Describe the solution you'd like**
Makefile could detect if it can use docker or podman

### PR DESCRIPTION
**Pull request by Elyytscha**
**What type of PR is this?**

/kind enhancement

**What does this PR do / why we need it**:

Adds Podman compatibility to Makefile

**Have you updated the necessary documentation?**

I think no doc update is needed for this

**Which issue(s) this PR fixes**:

Fixes #1298

**How to test changes / Special notes to the reviewer**:

```bash
make docker-build docker-push
make bundle-build bundle-push
make catalog-build catalog-push
```

with podman or docker installed, both will work

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,6 +5,14 @@
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
 VERSION ?= 0.9.0
 
+# Try to detect Docker or Podman
+CONTAINER_RUNTIME := $(shell command -v docker 2> /dev/null || command -v podman 2> /dev/null)
+
+# If neither Docker nor Podman is found, print an error message and exit
+ifeq ($(CONTAINER_RUNTIME),)
+$(error "No container runtime (Docker or Podman) found in PATH. Please install one of them.")
+endif
```
Comment (svghadi) [inline Makefile:14]: Can we remove this check or just log the message and continue without exiting? My concern with this is that if I just want to run some non image build command like `make run` or `make install` or some unit test on system without docker/podman, it won't allow me to do so.

**Review by svghadi — CHANGES_REQUESTED** [review]
Thank You @Elyytscha for the PR. I have a minor suggestion, other than that look good.

File: Makefile (hunk 1 above)
Comment (Elyytscha) [reply Makefile]: changed from error to warning, so now it should only print a warning instead of exiting with an error

**Review by svghadi — APPROVED** [review]

**Comment by Elyytscha** [comment]
feel free to squash it, if it gets merged, does not need to be 2 commits imo.

**Comment by svghadi** [comment]
Thanks @Elyytscha for the contribution.

**Pull request merged**